### PR TITLE
Mc userstories25 26

### PR DESCRIPTION
**Pull request by Mariana-21**
## All Submissions:

* [X] Merge the target branch into the PR branch. Fix any conflicts that might appear.
* [] Have you checked to ensure that other pull request are approved for merge [Pull Requests](../../../pulls)?
* [X] What stories does the PR include
  User stories 25-26
### New Feature Submissions:

* [X] Have you added an explanation of what your changes do?
* [X] Have you lint your code locally prior to submission?

### Testing:

* [X] Have you written new tests for your core changes, as applicable?
* [X] Have you successfully ran tests with your changes locally?
* [ ] All new and existing tests passed.

**Test coverage** = 98.58%
--------------------------------------------------------------------------------
### Change Log

* Added register/login links before checkout
* Added Test to assert registered user can checkout successfully 
* One older test fails, hence why we are at 98.59%, @alex-latham is going to go in a refactor.

### CODE REVIEW

File: app/controllers/cart_controller.rb [hunk 1]
```diff
@@ -13,6 +13,7 @@ def add_item
   end
 
   def show
+    @user = current_user
```
Comment (alex-latham) [inline app/controllers/cart_controller.rb]: Is `@user` used in the cart show html?

File: app/views/cart/show.html.erb [hunk 2]
```diff
@@ -37,18 +37,25 @@
...
+    <% end %>
+  </table>
+  <section id="cart-functions">
+    <p>Total: <%= number_to_currency(cart.total) %></p>
+    <p><%= link_to "Empty Cart", "/cart", method: :delete %></p>
+    <% if !current_user %>
+    <p> You need to register or login to checkout </p>
+    <p> <%= link_to "log in", "/login" %> </p>
+    <p> <%= link_to "register", "/register" %> </p>
+    <% else %>
```
Comment (alex-latham) [inline app/views/cart/show.html.erb]: US 25 says that the words "register" and "login" in the message should be links. Try:
`<p>You need to <%= link_to "register", "/register" %> or <%= link_to "log in", "/login" %> to checkout</p>`

Comment (Mariana-21) [reply app/views/cart/show.html.erb]: That worked!

File: app/controllers/cart_controller.rb (hunk 1 above)
Comment (Mariana-21) [reply app/controllers/cart_controller.rb]: I must have left it in there when I was trying to figure out the logic, good catch!

File: Gemfile [hunk 3]
```diff
@@ -5,6 +5,8 @@ git_source(:github) do |repo_name|
   "https://github.com/#{repo_name}.git"
 end
 
+Deprecate.skip = true if defined?(Deprecate.skip)
```
Comment (tylertomlinson) [inline Gemfile:8]: Slick

File: app/controllers/cart_controller.rb (hunk 1 above)
Comment (tylertomlinson) [reply app/controllers/cart_controller.rb]: Can you update PR

File: app/views/cart/show.html.erb [hunk 4]
```diff
@@ -2,26 +2,26 @@
...
+  <table class="cart-items">
+    <tr>
+      <th>Item</th>
+      <th>Sold By</th>
+      <th>Price</th>
+      <th>Quantity</th>
+      <th>Subtotal</th>
+    </tr>
+    <% @items.each do |item,quantity| %>
+    <tr id="cart-item-<%=item.id%>">
```
Comment (tylertomlinson) [inline app/views/cart/show.html.erb:14]: this indentation will need to be fixed eventually

File: app/views/cart/show.html.erb (hunk 2 above)
Comment (tylertomlinson) [reply app/views/cart/show.html.erb]: Update PR with what Alex said

File: spec/features/cart/registered_user_can_checkout_spec.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
+require 'rails_helper'
```
Comment (tylertomlinson) [inline spec/features/cart/registered_user_can_checkout_spec.rb:1]: Solid test

File: spec/models/order_spec.rb [hunk 6]
```diff
@@ -55,5 +55,9 @@
       expect(@item_order_2.status).to eq("unfulfilled")
       expect(@order_1.status).to eq("cancelled")
     end
+
+    it 'status' do
```
Comment (tylertomlinson) [inline spec/models/order_spec.rb:59]: Do the above excepts test this

File: Gemfile [hunk 7]
```diff
@@ -5,6 +5,8 @@ git_source(:github) do |repo_name|
...
+Gem::Deprecate.skip = true if defined?(Gem::Deprecate.skip)
```
Comment (danielpselph) [inline Gemfile:9]: Good use of this gem

**Review by danielpselph — APPROVED** [review]

**Review by alex-latham — APPROVED** [review]
Thanks for fixing those things!

**Review by tylertomlinson — APPROVED** [review]

**Pull request merged**